homeworks/08-functional-reactive-programming/src/App.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import {
+  combineLatest, from, interval, of, Subject,
+} from 'rxjs';
+import { useObservableState } from 'observable-hooks';
+import { map, scan } from 'rxjs/operators';
+import './App.css';
+
+import { Grid } from './components/Grid';
+
+import { generateGrid, getRandomTarget } from './utils';
+
+import { CellContent } from './types';
+
+export const App: React.FC = () => {
+  const [goodShots, setGoodShots] = useState<number>(0);
+
+  const shotTarget = new Subject<CellContent>();
+  const shotTarget$ = shotTarget.asObservable();
+
+  const targetIndex$ = interval(1000).pipe(map(() => getRandomTarget()));
+
+  const grid$ = of(generateGrid());
+
+  const cellsWithTarget$ = combineLatest([targetIndex$, grid$])
+    .pipe(
+      map(([targetIndex, grid]) => {
+        const newGrid = [...grid];
+        newGrid[targetIndex].content = CellContent.CAT;
+        return newGrid;
+      }),
+    );
+
+  const goodShots$ = from(shotTarget$)
```

Comment:
Observables should be inside a hook to avoid re-creating them.

Reply:
Created hook in https://github.com/vbarbasins/evo-ts-bootcamp/pull/9/commits/ab364b503dda4253ad1c53dadbf4500c5624acc7

And here added typing in hook and refactored code: https://github.com/vbarbasins/evo-ts-bootcamp/pull/9/commits/0019021647f6172788cb12977ae64e8043b8ec28